### PR TITLE
Improve local mode

### PR DESCRIPTION
**Pull request by MikeShi42**
Makes a number of improvements to local mode:
- Allows for custom port overwrites
- Hides irrelevant UI such as logout and log patterns in local mode
- Improve docs

General changes:
- Fix `NEXT_PUBLIC_SERVER_URL` typo in docker-compose (and standardize constant name exported from config)
- Make log patterns button more visible and more intuitive wording
- Allows for dynamic API server overrides without rebuilding the frontend (fixes https://github.com/hyperdxio/hyperdx/issues/12), though probably need some documentation.

### CODE REVIEW

File: docker/local/entry.sh [hunk 1]
```diff
@@ -37,6 +42,7 @@ ENABLE_GO_PARSER="false" \
 GO_PARSER_API_URL="http://go-parser:7777" \
 ENABLE_TOKEN_MATCHING_CHECK="false" \
 vector \
+  -qq \
```
Comment (wrn14897) [inline docker/local/entry.sh:45]: nit: imo it would be more readable to use `VECTOR_LOG`

File: packages/app/pages/_app.tsx [hunk 2]
```diff
@@ -125,13 +126,23 @@ export default function MyApp({ Component, pageProps }: AppPropsWithLayout) {
     fetch('/api/config')
       .then(res => res.json())
       .then(_jsonData => {
+        // Set API url dynamically for users who aren't rebuilding
+        try {
+          const url = new URL(_jsonData.apiServerUrl);
+          if (url != null) {
+            apiConfigs.prefixUrl = url.toString().replace(/\/$/, '');
```
Comment (wrn14897) [inline packages/app/pages/_app.tsx:133]: oh I see. So the hack is we reset this `apiConfigs` global var

File: packages/app/src/api.ts [hunk 3]
```diff
@@ -699,20 +719,20 @@ const api = {
   },
   useInstallation() {
     return useQuery<any, HTTPError>(`installation`, () =>
-      server(`installation`).json(),
+      hdxServer(`installation`).json(),
```
Comment (wrn14897) [inline packages/app/src/api.ts:722]: I assume the default method is GET

File: packages/app/pages/_app.tsx (hunk 2 above)
Comment (MikeShi42) [reply packages/app/pages/_app.tsx:133]: well the real hack is that we force everything to read the config on every request as opposed to having it pre-baked in the ky server instance hah

File: packages/app/src/PasswordResetPage.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ export default function PasswordResetPage({
     <div>
       <Form
         className="text-start"
-        action={`${API_SERVER_URL}/password-reset`}
+        action={`${SERVER_URL}/password-reset`}
```
Comment (wrn14897) [inline packages/app/src/PasswordResetPage.tsx:23]: double check that this url won't be updated dynamically, right ?

Comment (MikeShi42) [reply packages/app/src/PasswordResetPage.tsx:23]: Oh this is true, though this isn't important in local mode...

**Review by wrn14897 — APPROVED** [review]

File: packages/app/src/PasswordResetPage.tsx (hunk 4 above)
Comment (MikeShi42) [reply packages/app/src/PasswordResetPage.tsx:23]: We'll likely want to wrap this in a context and pull it from there in the future, but that's out of scope for this PR for now.

**Pull request merged**